internal/command/status/machines.go

```diff
@@ -81,18 +43,25 @@ func renderMachineStatus(ctx context.Context, app *api.AppCompact) (err error) {
 			machine.UpdatedAt,
 		})
 	}
-
-	_ = render.Table(io.Out, "", rows, "ID", "State", "Region", "Health checks", "Image", "Created", "Updated")
-
-	return
+	return render.Table(io.Out, "", rows, "ID", "State", "Region", "Health checks", "Image", "Created", "Updated")
 }
 
-func renderPGStatus(ctx context.Context, app *api.AppCompact, machines map[string]*api.Machine) (err error) {
-	var io = iostreams.FromContext(ctx)
-
+func renderPGStatus(ctx context.Context, app *api.AppCompact, machines []*api.Machine) (err error) {
+	io := iostreams.FromContext(ctx)
 	rows := [][]string{}
 
-	for role, machine := range machines {
+	for _, machine := range machines {
+		role := "unknown"
+		for _, check := range machine.Checks {
+			if check.Name == "role" {
+				if check.Status == "passing" {
+					role = check.Output
+				} else {
+					role = "failing"
```

Comment:
We usually print `error` when the role is not retrievable.

Reply:
No hard feelings on this one. I'll change to "error" if that fits better :)

Reply:
addressed by https://github.com/superfly/flyctl/pull/1308/commits/5c14a6940ef17b8a4a888bdd068bedbd77582abb